File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int, int> mp;
       
        for(int i = 0; i < nums.size(); i++){
            if(mp.find(target - nums[i]) == mp.end())
                mp[nums[i]] = i;
            else{
                int a = mp[target - nums[i]];
                int b = i;
                return {a, b};
            }
        }
 
        return {-1, -1};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mpp = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int need = target - nums[i];

            if (mpp.containsKey(need)) {
                int storedIndex = mpp.get(need);
                return new int[]{storedIndex, i};
            }

            mpp.put(nums[i], i);
        }

        return new int[]{-1, -1};
    }
}
